### ISSUE TITLE
enums not reflect in object response

### ISSUE THREAD

**Issue by trondhindenes** [piccolo-orm/piccolo#944]
Given this model:
```
class ReindexStatus(Enum):
    ready_to_start = 'ready_to_start'
    started = 'started'

class Test(Table):
  status = Varchar(choices=ReindexStatus, null=False)

```

if I do a query like 

```
thing = await Test.objects().first()
```
I would expect to be able to use the enum class to compare the object field value like:

```
if thing.status == ReindexStatus.ready_to_start:
```
However, the field seems to be returned just as a string.

This seems like a bug to me, is it intentional that enum-based fields aren't deserialized back to the enum?

**Comment by dantownsend** [comment]
I agree that converting the values back into enums would be a nice feature. I wonder if it should be a default behaviour, or configurable in some way (maybe via the `output` clause).

To make the enum comparable to the raw string value you can do this:

```python
# Note - it inherits from `str`
class ReindexStatus(str, Enum):
    ready_to_start = 'ready_to_start'
    started = 'started'
```

**Comment by trondhindenes** [comment]
I would argue that it should be the default value, since the `object()` call is an explicit desire to get the returned object back in a "fully deserialized" form, and Enum casting is part of that. Not sure if it would break for existing users tho.

Thanks for the trick, I'll play with that!

**Comment by juliushuck** [comment]
There is also `from enum import StrEnum`
